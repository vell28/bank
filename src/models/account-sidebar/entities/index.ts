export interface IAccountSidebar {
  [key: string]: boolean;
}
