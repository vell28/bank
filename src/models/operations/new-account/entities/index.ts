export enum AccountTypes {
  LINKED = 'LINKED',
  BUSINESS = 'BUSINESS',
  SAVINGS = 'SAVINGS',
}

export type AccountTypeType =
  | AccountTypes.LINKED
  | AccountTypes.BUSINESS
  | AccountTypes.SAVINGS;

export enum AccountNames {
  LINKED = 'Linked',
  BUSINESS = 'Business',
  SAVINGS = 'Savings',
}

export type AccountNameType =
  | AccountNames.LINKED
  | AccountNames.BUSINESS
  | AccountNames.SAVINGS;

export interface IAccount {
  type: AccountTypeType;
  name: AccountNameType;
}

export const ACCOUNTS: IAccount[] = [
  {
    type: AccountTypes.LINKED,
    name: AccountNames.LINKED,
  },
  {
    type: AccountTypes.BUSINESS,
    name: AccountNames.BUSINESS,
  },
  {
    type: AccountTypes.SAVINGS,
    name: AccountNames.SAVINGS,
  },
];
