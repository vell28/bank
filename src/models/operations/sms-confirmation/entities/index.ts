export interface IConfirmation {
  confirmationCodeLength?: number;
  confirmationToken?: string;
  attempts?: number;
}
