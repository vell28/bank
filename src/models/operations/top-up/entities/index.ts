export interface ICardData {
  cardNumber?: string;
  beneficiaryName?: string;
  expiresDate?: string;
  isHolder?: boolean;
  cvv?: string;
  country?: string;
  city?: string;
}

export interface ITopUpRequest {
  email?: string;
  initiatorName?: string;
}

export interface ITopUp {
  card: ICardData;
  value: string;
  redirectUrl?: string;
  description?: string;
  request: ITopUpRequest;
}
