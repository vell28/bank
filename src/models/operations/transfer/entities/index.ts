export interface ICardData {
  cardNumber?: string;
  beneficiaryName?: string;
  expiresDate?: string;
  cvv?: string;
}

export interface IBankData {
  beneficiaryName?: string;
  iban?: string;
  country?: string;
  city?: string;
  address?: string;
  swiftCode?: string;
  beneficiaryBank?: string;
  isIntermediaryBank?: boolean;
  intermediarySwift?: string;
  intermediaryBank?: string;
  intermediaryAccount?: string;
  commissionType?: SwiftCommissionType;
  urgency?: SwiftUrgencyType;
  purpose?: string;
  transferDetails?: string;
}

export interface IRate {
  currencyFrom: string;
  currencyTo: string;
  fromAmount2: number;
  fromAmount3:number;
  fromAmount4: number;
  rate: number;
  rate2: number;
  rate3: number;
  rate4: number;
}

export type IRates = IRate | [];

export interface ITransfer {
  card: ICardData;
  phone: string;
  value: string;
  purpose: string;
  bank: IBankData;
  rates: IRates;
}

export enum SwiftCommissionTypes {
  BENEFICIARY = 'BENEFICIARY',
  SHARED = 'SHARED',
  SENDER = 'SENDER',
}

export type SwiftCommissionType =
  | SwiftCommissionTypes.BENEFICIARY
  | SwiftCommissionTypes.SHARED
  | SwiftCommissionTypes.SENDER;

export enum SwiftUrgencyTypes {
  NORMAL = 'Normal',
  URGENT = 'Urgent',
}

export type SwiftUrgencyType = SwiftUrgencyTypes.NORMAL | SwiftUrgencyTypes.URGENT;
