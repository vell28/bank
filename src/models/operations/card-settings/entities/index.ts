import { CurrencyCodeType } from 'modules/currencies';
import { AccountStatusType, IBalance } from 'models/organizations/entities';

export enum CardProduct {
  MAIN = 'MAIN',
  ADDITIONAL = 'ADDITIONAL',
  FAMILY = 'FAMILY',
  KIDS = 'KIDS',
}

export type CardProductType = CardProduct.MAIN | CardProduct.ADDITIONAL | CardProduct.FAMILY | CardProduct.KIDS;

export interface ICardOwner {
  embossedName: string;
}

export enum Limit {
  DAY = 'DAY',
  MONTH = 'MONTH',
  FREE_ATM_WITHDRAWAL = 'FREE_ATM_WITHDRAWAL',
}

export type LimitType = Limit.DAY | Limit.MONTH | Limit.FREE_ATM_WITHDRAWAL;

export interface ILimit {
  type: LimitType;
  title: null | string;
  maxValue: number;
  usedValue: number;
  preferredMaxValues?: number[];
}

export interface ILimitData {
  type: LimitType;
  maxValue: string;
}

export interface ICardOptions {
  limits?: {
    disable?: boolean;
    disable5min?: boolean;
    disableOneTime?: boolean;
  };
}

export interface ICard {
  accountId: string;
  availableBalance: number;
  backgroundImage?: string;
  balance: number;
  balanceHold: number;
  balances: IBalance[];
  clientId: string;
  createdAt?: string;
  currency: {
    code: CurrencyCodeType;
  };
  expireAt?: string;
  id: string;
  limits: ILimit[];
  number?: string;
  options: ICardOptions;
  owner: ICardOwner;
  paySystem: string;
  plasticStatus: string;
  productType: CardProductType;
  status: AccountStatusType;
  title?: string;
}

export interface ICardUnmasked {
  number: string;
  pin: string | null;
  cvc: string;
  expireAt: string;
  owner: ICardOwner;
}
