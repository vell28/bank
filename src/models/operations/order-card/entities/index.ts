import { CardProductType, ICardOwner } from '../../card-settings/entities/index';

export interface ICardOrderOwner extends ICardOwner {
  phoneNumber: string;
}

export interface IAddress {
  apartmentNumber: number;
  city: string;
  countryCode: string;
  postalCode: string;
  province?: string;
  street: string;
  streetNumber: string;
}

export interface ICardOrder {
  cardProductType?: CardProductType;
  cardOwner?: ICardOrderOwner;
  deliveryAddress?: IAddress;
}

export enum CardFormat {
  VIRTUAL = 'VIRTUAL',
  PLASTIC = 'PLASTIC',
}

export type CardFormatType = CardFormat.VIRTUAL | CardFormat.PLASTIC;

export interface IOrderRequestBody {
  type: CardProductType;
  format: CardFormatType;
  cardHolderName: string;
  cardHolderPhoneNumber: string;
  isExpressDelivery: boolean;
  deliveryAddress: IAddress;
}
