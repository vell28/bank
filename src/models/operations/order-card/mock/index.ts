import { IServerAddress } from 'models/organizations/redux/actions/entities';
import { IClient } from 'models/organizations/entities';
import { IAddress, ICardOrderOwner } from '../entities';

export const exampleAddress: IAddress = {
  countryCode: 'DE',
  city: 'Cologne',
  postalCode: '50667',
  street: 'Domkloster',
  streetNumber: '4',
  apartmentNumber: 0,
};

export const exampleOrganizationAddress: IAddress = {
  countryCode: 'DE',
  city: 'Cologne',
  postalCode: '50667',
  street: 'Domkloster',
  streetNumber: '4',
  apartmentNumber: 0,
};

export const exampleClientInfo: ICardOrderOwner = {
  embossedName: 'Frank-Walter Steinmeier',
  phoneNumber: '9111111',
};

export const exampleServerAddress: IServerAddress = {
  country: 'Germany',
  province: 'North Rhine-Westphalia',
  city: 'Düsseldorf',
  postalCode: '40479',
  street: 'Gartenstraße',
  streetNumber: '53',
  apartmentNumber: '42',
};

export const exampleClient: IClient = {
  address: exampleOrganizationAddress,
  name: 'Frank-Walter Steinmeier',
  phoneNumber: '9111111',
};
