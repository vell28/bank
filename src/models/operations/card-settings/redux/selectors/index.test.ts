import { accountsData, cardData } from 'models/organizations/mock';
import { emptyStore } from 'modules/store/emptyStore';
import { IStore } from 'modules/store/types';
import {
  getUnmaskedCardById,
  getFullUnmaskedSelectedCard,
  getSliderActiveIndex,
  getSliderActiveCard,
  getIsShownLockModal,
  getIsShownUnLockConfirmModal,
  getIsShownUnLockModal,
  getIsShownDisableLimitsModal,
  getIsShownDisableLimitsConfirmModal,
  getIsShownIncreaseLimitModal,
  getIsShownIncreaseLimitConfirmModal,
  getIncreaseLimit,
  getIsShownDisableLimitsTimer,
  getCardLimitsDisable,
} from '.';
import { Limit } from '../../entities';

const unMasked = {
  number: '**** **10 0004 ****',
  pin: null,
  cvc: '000',
  expireAt: '2022-07-01T00:00:00',
  owner: {
    embossedName: 'Card holder',
  },
};

const store: IStore = {
  ...emptyStore,
  organizations: {
    isLoading: false,
    isSuccess: false,
    isError: false,
    data: [
      {
        clientName: 'Ralf Williams',
        id: 1,
        name: 'papaya',
        tin: 'PAPYMTMTXXX',
        title: 'Papaya Ltd',
        accounts: accountsData,
        cards: [cardData],
        client: {},
      },
    ],
    errors: [],
  },
  operations: {
    ...emptyStore.operations,
    cardSettings: {
      isLoading: false,
      isSuccess: false,
      isError: false,
      data: { [cardData.id]: unMasked },
      selectedCard: {
        [accountsData[0].id]: {
          card: cardData,
          index: 5,
        },
      },
      modals: {
        isShownLock: true,
        isShownUnLock: true,
        isShownUnLockConfirm: true,
        isShownDisableLimits: true,
        isShownDisableLimitsConfirm: true,
        isShownIncreaseLimit: true,
        isShownIncreaseLimitConfirm: true,
      },
      errors: [],
      limitsSettings: {
        isShownDisableLimitsTimer: true,
        increasedLimit: { type: Limit.DAY, maxValue: '1000' },
      },
    },
  },
  router: {
    action: 'PUSH',
    location: {
      pathname: `/banking/account/${accountsData[0].id}/EUR`,
      search: '',
      hash: '',
      key: '',
      state: '',
    },
  },
};

describe('unmasked card selector test', () => {
  test('getUnmaskedCardById should return unmasked card || undefined', () => {
    const card = getUnmaskedCardById(cardData.id)(store);
    expect(card).toBe(unMasked);

    const anotherCard = getUnmaskedCardById('another')(store);
    expect(anotherCard).toBeUndefined();
  });

  test('getFullUnmaskedSelectedCard should return card with unmasked number', () => {
    const card = getFullUnmaskedSelectedCard(store);
    expect(card.number.indexOf('*')).toBe(-1);
    expect(card).toMatchSnapshot();
  });

  test('getSliderActiveIndex should return active slider index by Account', () => {
    const index = getSliderActiveIndex(accountsData[0].id)(store);
    expect(index).toEqual(5);
  });
  test('getSliderActiveCard should return active card in slider', () => {
    const card = getSliderActiveCard(store);
    expect(card).toEqual(cardData);
  });
  test('getIsShownLockModal should return true', () => {
    const isShown = getIsShownLockModal(store);
    expect(isShown).toBeTruthy();
  });
  test('getIsShownUnLockConfirmModal should return true', () => {
    const isShown = getIsShownUnLockConfirmModal(store);
    expect(isShown).toBeTruthy();
  });
  test('getIsShownUnLockModal should return true', () => {
    const isShown = getIsShownUnLockModal(store);
    expect(isShown).toBeTruthy();
  });
  test('getIsShownDisableLimitsModal should return true', () => {
    const isShown = getIsShownDisableLimitsModal(store);
    expect(isShown).toBeTruthy();
  });
  test('getIsShownDisableLimitsConfirmModal should return true', () => {
    const isShown = getIsShownDisableLimitsConfirmModal(store);
    expect(isShown).toBeTruthy();
  });
  test('getIsShownIncreaseLimitModal should return true', () => {
    const isShown = getIsShownIncreaseLimitModal(store);
    expect(isShown).toBeTruthy();
  });
  test('getIsShownIncreaseLimitConfirmModal should return true', () => {
    const isShown = getIsShownIncreaseLimitConfirmModal(store);
    expect(isShown).toBeTruthy();
  });
  test('getIncreaseLimit should return limit', () => {
    const limit = getIncreaseLimit(store);
    expect(limit).toEqual({ type: 'DAY', maxValue: '1000' });
  });
  test('getIsShownDisableLimitsTimer should return true', () => {
    const isShown = getIsShownDisableLimitsTimer(store);
    expect(isShown).toBeTruthy();
  });
  test('getCardLimitsDisable should return false', () => {
    const isDisable = getCardLimitsDisable(store);
    expect(!isDisable).toBeTruthy();
  });
});
