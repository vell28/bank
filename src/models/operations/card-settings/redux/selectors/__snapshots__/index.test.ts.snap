// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unmasked card selector test getFullUnmaskedSelectedCard should return card with unmasked number 1`] = `
Object {
  "cvc": "000",
  "expireAt": "2022-07-01T00:00:00",
  "number": "5277 6010 0004 3984",
  "owner": Object {
    "embossedName": "Card holder",
  },
  "pin": null,
}
`;
